### ISSUE TITLE
Web5 Quickstart doesn't yet deliver for users

### ISSUE THREAD

**Issue by ALRubinger** [TBD54566975/developer.tbd.website#398]
We have content and tech work to do here.

The Quickstart must address the *why* and the *how* of Web5.

* Quickstart should be able to be completed as promised in in 5min 
* Starts from a template
* Nails the "why" of Web5 and a Personal Data Store for users of your (as an app dev) app 

## The Why

What is the value prop of Web5? Why a paradigm shift into a personal data store? How is this a fundamental building block for making a whole new class of decentralized applications that preserve users' privacy?

Right now the message is centered on the CRUD operations involved.

## The How

I don't believe the Quickstart in its current form can get the user what's promised in 5min. Have any of us on the team run through this to try it out? I did this evening, and I had to:

* Pull from prior knowledge in software development
* Use other tools to bootstrap a project; we don't mention how to make this new project in the Quickstart
* Refer to Devin's full example app quite a bit; the code we put in the Quickstart is only the Web5-specific code; the user is left to figure out if they should put these in functions, and how to display the return values on the page. 
* ...and this all took me much longer than 5min.

The Web5 Quickstart starts with an installation section:

```
npm install @tbd54566975/web5
```

This assumes the user has a current project, and that they're using NPM. How is a new user to create this project? What assumptions have we made about that user's expertise with coding and their ability to start a new project? (This will be handled by #408)

In short, following the instructions on the Quickstart exactly as written does not give the user everything they need to get a working application. Some of it can be pieced together through prior knowledge and by referring to Devin's fully working application example -- but there are omissions in the Quickstart instructions which may put a user in a dead end. There are no functions mentioned; no HTML for input fields or writing return values.

I'd be surprised if anyone on the team can follow the Quickstart and get a working Web5 application in 5 minutes as promised. And this has to work for relative newcomers to front end coding; assume the requisite experience to meet the "5min" bar is the equivalent of one semester of software development coursework in JavaScript.

**Comment by nearlyjuly** [comment]
I think there's been some duplication of work in creating all these great demos. There wouldn't really be a Web5 app, that would be a DWA that works with Web5 components. The DWA building docs aren't ready yet according to https://areweweb5yet.com/ There are several ways that a DWN can be utilised and set up. The dwn-sdk is for creating any type of DWN. The Web5-js can create a personal datastore DWN for testing (in the browser or running on a local computer). The dwn-js repo will also set up a personal datastore DWN for testing, and a more robust version is in dwn-server with some hosted DWNs available for testing. What Web5-js does is provide a simple way to talk to a DWN (any type) as this is time consuming to code the queries routing manually. It can create DIDs but there is a separate ecosystem for storing and managing these (wallets, SSI repos). You would use dwn-sdk to make a reusable one, and there is the upcoming Electron standalone DWN coming soon. 

So maybe it's better to see this as a 5 minute Quickstart to communicate using Web5, and use a hosted DWN (with the tenant locked to the owner to stop anyone else writing to it). The visitor to this page then creates a DID, sends a message to the hosted DWN by a provided DID and then a message is returned to say something like congrats or well done. Then the visitor can at least know what a DID is and how it can be used instead of a server address to access someone else's info.

**Comment by angiejones** [comment]
also see my comments

https://github.com/TBD54566975/developer.tbd.website/issues/392#issuecomment-1507997361

**Comment by ALRubinger** [comment]
@csuwildcat recommending @dcrousso may work on a "starter template", based on this:

https://github.com/pwa-builder/pwa-starter

That'd give new users a starting point to Web5-ize from.

**Comment by dayhaysoos** [comment]
I guess the best thing we can do to close this out is find someone to run through these steps and see how far they get?

I think the starter template @ALRubinger would be really helpful. 

I also can't help but feel like no one would actually want to use this quick start to continue building a whole app. These days, I personally would reach for something like Next.js or Vite. I think that we should consider having templates based on things like that.

**Comment by ALRubinger** [comment]
@dayhaysoos Spot on - and @csuwildcat has elevated "starter template" for Phase 3. I agree that'll be really helpful - removes all the need for:

1) Us to explain boilerplate and add to conceptual weight of the Quickstart
2) Reader to have to spend time making boilerplate.

A clean starting point for them plus a Quickstart that's about how to Web5-ize it is a great flow for now. And opens us up to an app to get people started later. For now templates are great.

**Comment by ALRubinger** [comment]
Slating as TODO for Phase 3 and self-assigning; open to pairing w/ interested/available parties on it. @chris-tbd @bobbilee19 @angiejones @EbonyLouis

**Comment by ALRubinger** [comment]
Template to help with the first setup is in #408

**Comment by ALRubinger** [comment]
@bobbilee19 and I gonna pair on driving this one to the finish :)

**Comment by ALRubinger** [comment]
@bobbilee19 and I will center this work in `upstream` branch `issue-398-quickstart-upgrades`

**Comment by ALRubinger** [comment]
Deploy preview for the branch noted above is [here](https://deploy-preview-419--tbd-website-developer.netlify.app/docs/web5/quickstart).

**Comment by dayhaysoos** [comment]
@bobbilee19 has come across some errors during the upgrade process. We looped in OSE to get help on this which allowed us to get through a few errors but we need to reconvene to wrap things up.

**Comment by ALRubinger** [comment]
Encompassed by a series of changes done coordinated @bobbilee19, advised by @angiejones and I, and now in `main`.
